abrams_strogatz/lang_comp_bilinguals.py

```diff
@@ -0,0 +1,300 @@
+    p(AB->A) = s * (nA + nAB)**a
+    The probability to change from language AB to B is:
+    p(B->AB) = (1 - s) * (nB + nAB)**a
+    The steps are the following:
+    1) A random node is selected.
+    2) The language of each neighbor surrounding the node is computed,
+    counting the number of speakers.
+    3) The probability of change is computed. If the probability is
+    bigger than a uniformly distributed random number, the language
+    of the selected node is changed.
+
+    Args:
+        popu: Array containing the language spoken by the population.
+            The values contained inside the array are 1 (lang A), 
+            0 (bilingual) and -1 (lang B).
+        m: Size of the array.
+        n: Size of the array.
+        s: Status or prestige of the language A.
+        a: Volatility of the system. Determines the location of the
+            fixed points. Defaults to 1.0
+
+    Returns:
+        The function returns the updated version of the np.array popu.
+    """
+    # Selection random node.
+    ii, jj = np.random.randint(m), np.random.randint(n)
+    lang = popu[ii, jj]
+    # Neighbors language
+    nn = np.array([])  # Array containing the neighbors
+    nn = np.append(nn, popu[periodic_boundary((ii - 1, jj), (m, n))])  # upper
+    nn = np.append(nn, popu[periodic_boundary((ii + 1, jj), (m, n))])  # lower
+    nn = np.append(nn, popu[periodic_boundary((ii, jj - 1), (m, n))])  # left
+    nn = np.append(nn, popu[periodic_boundary((ii, jj + 1), (m, n))])  # right
+    # Number of speakers
+    nA = (nn == 1.0).sum() / 4.0   # Number A speakers
+    nB = (nn == -1.0).sum() / 4.0  # Number B speakers
+    nAB = 1.0 - nA - nB            # Number AB speakers 
+    # Language dynamics
+    # If lang = 1 => prob(A->AB). If lang = -1 => prob(B->AB). If lang = 0 => prob(AB->A) and
+    # prob(AB->B)
+    if lang == 1:
+        popu[ii, jj] = 0 if (np.random.uniform() < ((1.0 - s) * nB ** a)) else popu[ii, jj]
+    elif lang == -1: 
+        popu[ii, jj] = 0 if (np.random.uniform() < (s * nA ** a)) else popu[ii, jj]
+    else:
+        prob_change1 = s * (nA + nAB) ** a          # Change AB -> A 
+        prob_change2 = (1 - s) * (nB + nAB) ** a    # Change AB -> B
+        u, v = np.random.uniform(), np.random.uniform()
+        if (u > prob_change1) and (v > prob_change2):
+            popu[ii, jj]    # The condition is not fulfilled. 
+        else:
+            # Can occur that both p(AB->A) and p(AB->B) are satisfied at the same time. 
+            # We have to compute again the probability of change until only one of the conditions
+            # is satisfied. 
+            while (u < prob_change1) and (v < prob_change2):
+                u, v = np.random.uniform(), np.random.uniform()
+            popu[ii, jj] = 1 if (u < prob_change1) else -1
+
+    return popu
+
+
+def saving_process(popu: np.array, ph: int, key: bool = True):
+    """
+    Function used to save language grid plots.
```

Comment:
177:   1 D401 First line should be in imperative mood; try rephrasing [flake8-docstrings]
```suggestion
    Generate and save language grid plots.
```